src/user-util/index.ts

```diff
@@ -45,6 +45,102 @@ ${el}`;
   }
 };
 
+// Why is this hardcoded?
+//   So that the snapshots do not fail when a new version is released and all the error messages change
+// Why is this not pointing to `main`?
+//   So that if the docs are moved around or renamed in the future, the links in previous PT versions still work
+// Does this need to be updated before every release?
+//   No, only when the docs are changed
+const docsVersion = 'v2.0.0';
+const customizeDocsMessage = `You can customize this check by setting the targetSize option, more details at https://github.com/cloudfour/pleasantest/blob/${docsVersion}/docs/errors/target-size.md`;
+const errorMessage = 'Cannot click element that is too small.';
+
+type TargetSize = number | true | undefined;
+
+export const assertTargetSize = (el: Element, targetSize: TargetSize) => {
+  // Per W3C recommendation, inline elements are excluded from a min target size
+  // See: https://www.w3.org/WAI/WCAG21/Understanding/target-size.html
+  if (getComputedStyle(el).display === 'inline') {
+    return;
+  }
+
+  const { width, height } = el.getBoundingClientRect();
+  const size = getTargetSize(targetSize);
+
+  if (width < size || height < size) {
+    // Custom messaging for inputs that should have labels (e.g. type="radio").
+    //
+    // Inputs that aren't expected to have labels (e.g. type="submit") are
+    // checked by the general element check.
+    //
+    // MDN <input> docs were referenced and the following were assumed
+    // to not have labels:
+    // - type="submit"
+    // - type="button"
+    // - type="reset"
+    //
+    // @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input
+    if (
+      el instanceof HTMLInputElement &&
+      el.type !== 'submit' &&
+      el.type !== 'button' &&
+      el.type !== 'reset'
+    ) {
+      return checkInputEl({ el, targetSize });
+    }
+
+    // General element messaging
+    throw error`${errorMessage}
+${getTargetSizeMessage({ el, targetSize })}
+${getDimensionsMessage(el)}
+${el}
+${customizeDocsMessage}`;
+  }
+};
+
+/** Handles inputs that should have labels */
+const checkInputEl = ({
+  el,
+  targetSize,
+}: {
+  el: HTMLInputElement;
+  targetSize: TargetSize;
+}) => {
+  const labelSize = el.labels?.[0]?.getBoundingClientRect();
+  const size = getTargetSize(targetSize);
+
+  // Element did not have label
+  if (!labelSize) {
+    throw error`${errorMessage}
+${getTargetSizeMessage({ el, targetSize })}
+${getDimensionsMessage(el)}
+${el}
+You can increase the target size of the ${getElementDescriptor(
+      el,
+    )} by adding a label that is larger than ${size}px × ${size}px
+${customizeDocsMessage}`;
+  }
+
+  // If label is valid
+  if (labelSize.width >= size && labelSize.height >= size) {
+    return;
+  }
+
+  // Element and label was too small
+  throw error`${errorMessage}
+${getTargetSizeMessage({ el, targetSize })}
+${getDimensionsMessage(el)}
+${el}
```

Comment:
@gerardo-rodriguez I'm noticing that these error messages have more in common than different so mind if I rearrange a little bit to call a single function with the element and the "suggestion text", where the function would construct the error message?

Reply:
@calebeby Go for it! 👍